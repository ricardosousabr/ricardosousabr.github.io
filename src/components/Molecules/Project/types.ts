export interface IProjectProps {
  title: string;
  subtitle: string;
  button: string;
  projectsCreated: {
    name: string;
    link: string;
    srcMobile: string;
    srcDesktop: string;
    technologies: string[];
  }[];
}