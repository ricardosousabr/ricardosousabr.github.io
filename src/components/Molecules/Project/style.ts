import styled from "styled-components";
import screens from "../../../tokens/screen";

export const ProjectStyle = styled.div`
  background: #111827;
  display: flex;
  flex-direction: column;
  text-align: center;
  justify-content: space-between;
  padding: 50px 0;

  @media (min-width: ${screens.extraLarge}) {
    padding: 50px 50px;

  }

`

export const BoxProject = styled.div`
  padding-bottom: 30px;
`

export const BoxText = styled.div`
  display: flex;
  align-items: center;
  height: 60px;
  justify-content: space-between;
  flex-direction: column;
  padding-bottom: 50px;
`

export const BoxProjects = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;

  @media (min-width: ${screens.extraLarge}) {
    flex-direction: row;
    justify-content: space-between;

  }

`
export const BoxImage = styled.div`
  width: 250px;
  height: 300px;
  padding-bottom: 10px;

  @media (min-width: ${screens.small}) {
    width: 400px;
    height: 250px;
    border-radius: 10px;

  }

`

export const BoxImageMobile = styled.div`
    @media (min-width: ${screens.small}) {
    display: none;
  }

`

export const BoxImageDesktop = styled.div`
    display: none;
    @media (min-width: ${screens.small}) {
    display: block;

    }

`
