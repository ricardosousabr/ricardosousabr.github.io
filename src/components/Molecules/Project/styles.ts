import styled from "styled-components";
import screensYoga from "../../../tokens/screen";

export const Wrapper = styled.div`
  background: #282C33;
  padding: 50px 0;

`

export const ProjectStyle = styled.div`
  display: flex;
  flex-direction: column;
  text-align: center;
  justify-content: space-between;
  text-align: center;

`

export const BoxTitle = styled.div`
  text-align: center;
  padding-bottom: 50px;

`

export const Title = styled.div`
  padding-bottom: 20px;

`

export const BoxProjects = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;

  @media (min-width: ${screensYoga.breakpointXl}) {
    flex-direction: row;
    justify-content: space-between;
  }

`

export const BoxImage = styled.div`
  max-width: 350px;

`

export const BoxImageMobile = styled.div`
    border-bottom: 1px solid #ABB2BF;

  @media (min-width: ${screensYoga.breakpointSm}) {
    display: none;
  }

`

export const BoxImageDesktop = styled.div`
  display: none;
  border-bottom: 1px solid #ABB2BF;

  @media (min-width: ${screensYoga.breakpointSm}) {
    display: block;
  }

`

export const BoxTechnologies = styled.div`
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  max-width: 250px;

  @media (min-width: ${screensYoga.breakpointMD}) {
    max-width: 350px;
  }

`

export const BoxTechnologie = styled.div`
  display: flex;
  flex-wrap: wrap;
  padding: 10px 0;
  margin: 0 8px;
`

export const BoxText = styled.div`
  border-top: 1px solid #ABB2BF;
  text-align: center;
  padding: 10px 0;
`