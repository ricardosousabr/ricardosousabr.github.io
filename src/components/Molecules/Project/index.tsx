import Image from "next/image";
import Text from "../../Atoms/Text";
import Button from "../../Atoms/Button";
import Container from "../../Atoms/Container";
import { Wrapper, ProjectStyle, BoxProjects, BoxProject, BoxImage, BoxImageMobile, BoxImageDesktop, BoxText } from "./style";

export default function Project() {
  return (
    <Wrapper>
      <Container>
        <ProjectStyle>
          <BoxText>
            <Text color="white" fontSize="20px">Project</Text>
            <Text color="white" fontSize="16px">Look at some of my projects</Text>
          </BoxText>
          <BoxProjects>
            <BoxProject>
              <BoxImage>
                <BoxImageMobile>
                  <Image width={250} height={300} src="/img/Project-Login-Page-Mobile.png" alt="" />
                </BoxImageMobile>
                <BoxImageDesktop>
                  <Image width={350} height={250} src="/img/Project-Login-Page-Desktop.png" alt="" />
                </BoxImageDesktop>
              </BoxImage>
              <Button color="white" padding="0px" background="transparent" borderRadius="0px" border="none" fontSize="16px" textDecoration="none" href="https://ricardosousabr.github.io/login-page-next/">Login page next </Button>
            </BoxProject>
            <BoxProject>
              <BoxImage>
                <BoxImageMobile>
                  <Image width={250} height={300} src="/img/Project-Pokedex-Mobile.png" alt="" />
                </BoxImageMobile>
                <BoxImageDesktop>
                  <Image width={350} height={250} src="/img/Project-Pokedex-Desktop.png" alt="" />
                </BoxImageDesktop>
              </BoxImage>
              <Button color="white" padding="0px" background="transparent" borderRadius="0px" border="none" fontSize="16px" textDecoration="none" href="https://ricardosousabr.github.io/login-page-next/">Pokedex </Button>
            </BoxProject>
            <BoxProject>
              <BoxImage>
                <BoxImageMobile>
                  <Image width={250} height={300} src="/img/Project-Storybook-Mobile.png" alt="" />
                </BoxImageMobile>
                <BoxImageDesktop>
                  <Image width={350} height={250} src="/img/Project-Storybook-Desktop.png" alt="" />
                </BoxImageDesktop>
              </BoxImage>
              <Button color="white" padding="0px" background="transparent" borderRadius="0px" border="none" fontSize="16px" textDecoration="none" href="https://ricardosousabr.github.io/login-page-next/">Storybook </Button>
            </BoxProject>
          </BoxProjects>
        </ProjectStyle>
      </Container>
    </Wrapper>
  )
}