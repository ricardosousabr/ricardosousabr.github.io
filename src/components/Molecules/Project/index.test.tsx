import renderer from "react-test-renderer";
import Project from "./index";
import locale from "../../../locales";

test("Link renders correctly", () => {
  const { project } = locale['pt-br'];
  const tree = renderer.create(<Project {...project} />).toJSON();
  expect(tree).toMatchSnapshot();
});
