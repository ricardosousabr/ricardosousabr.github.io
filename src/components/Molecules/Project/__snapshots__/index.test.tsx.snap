// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders correctly 1`] = `
<div
  className="sc-hKFymg kUSEzG"
>
  <div
    className="sc-dlniIP iETDvP"
  >
    <div
      className="sc-eCAqax dXoDOC"
    >
      <div
        className="sc-gKAaef cFLQtP"
      >
        <p
          className="sc-bdnyFh iOsQJx"
          color="white"
          fontSize="20px"
        >
          Project
        </p>
        <p
          className="sc-bdnyFh iOusbc"
          color="white"
          fontSize="16px"
        >
          Look at some of my projects
        </p>
      </div>
      <div
        className="sc-iCoHzw eApTuQ"
      >
        <div
          className="sc-jSFipO iUydkh"
        >
          <div
            className="sc-fujznN gmeAfE"
          >
            <div
              className="sc-pNWRh kqFoPp"
            >
              <img
                alt=""
                data-nimg="1"
                decoding="async"
                height={300}
                loading="lazy"
                onError={[Function]}
                onLoad={[Function]}
                src="/_next/image?url=%2Fimg%2FProject-Login-Page-Mobile.png&w=640&q=75"
                srcSet="/_next/image?url=%2Fimg%2FProject-Login-Page-Mobile.png&w=256&q=75 1x, /_next/image?url=%2Fimg%2FProject-Login-Page-Mobile.png&w=640&q=75 2x"
                style={
                  {
                    "color": "transparent",
                  }
                }
                width={250}
              />
            </div>
            <div
              className="sc-jrsKJM cQLIgn"
            >
              <img
                alt=""
                data-nimg="1"
                decoding="async"
                height={250}
                loading="lazy"
                onError={[Function]}
                onLoad={[Function]}
                src="/_next/image?url=%2Fimg%2FProject-Login-Page-Desktop.png&w=750&q=75"
                srcSet="/_next/image?url=%2Fimg%2FProject-Login-Page-Desktop.png&w=384&q=75 1x, /_next/image?url=%2Fimg%2FProject-Login-Page-Desktop.png&w=750&q=75 2x"
                style={
                  {
                    "color": "transparent",
                  }
                }
                width={350}
              />
            </div>
          </div>
          <a
            background="transparent"
            border="none"
            borderRadius="0px"
            className="sc-gtsqUy jKDWbH"
            color="white"
            fontSize="16px"
            href="https://ricardosousabr.github.io/login-page-next/"
            padding="0px"
            textDecoration="none"
          >
            Login page next 
          </a>
        </div>
        <div
          className="sc-jSFipO iUydkh"
        >
          <div
            className="sc-fujznN gmeAfE"
          >
            <div
              className="sc-pNWRh kqFoPp"
            >
              <img
                alt=""
                data-nimg="1"
                decoding="async"
                height={300}
                loading="lazy"
                onError={[Function]}
                onLoad={[Function]}
                src="/_next/image?url=%2Fimg%2FProject-Pokedex-Mobile.png&w=640&q=75"
                srcSet="/_next/image?url=%2Fimg%2FProject-Pokedex-Mobile.png&w=256&q=75 1x, /_next/image?url=%2Fimg%2FProject-Pokedex-Mobile.png&w=640&q=75 2x"
                style={
                  {
                    "color": "transparent",
                  }
                }
                width={250}
              />
            </div>
            <div
              className="sc-jrsKJM cQLIgn"
            >
              <img
                alt=""
                data-nimg="1"
                decoding="async"
                height={250}
                loading="lazy"
                onError={[Function]}
                onLoad={[Function]}
                src="/_next/image?url=%2Fimg%2FProject-Pokedex-Desktop.png&w=750&q=75"
                srcSet="/_next/image?url=%2Fimg%2FProject-Pokedex-Desktop.png&w=384&q=75 1x, /_next/image?url=%2Fimg%2FProject-Pokedex-Desktop.png&w=750&q=75 2x"
                style={
                  {
                    "color": "transparent",
                  }
                }
                width={350}
              />
            </div>
          </div>
          <a
            background="transparent"
            border="none"
            borderRadius="0px"
            className="sc-gtsqUy jKDWbH"
            color="white"
            fontSize="16px"
            href="https://ricardosousabr.github.io/login-page-next/"
            padding="0px"
            textDecoration="none"
          >
            Pokedex 
          </a>
        </div>
        <div
          className="sc-jSFipO iUydkh"
        >
          <div
            className="sc-fujznN gmeAfE"
          >
            <div
              className="sc-pNWRh kqFoPp"
            >
              <img
                alt=""
                data-nimg="1"
                decoding="async"
                height={300}
                loading="lazy"
                onError={[Function]}
                onLoad={[Function]}
                src="/_next/image?url=%2Fimg%2FProject-Storybook-Mobile.png&w=640&q=75"
                srcSet="/_next/image?url=%2Fimg%2FProject-Storybook-Mobile.png&w=256&q=75 1x, /_next/image?url=%2Fimg%2FProject-Storybook-Mobile.png&w=640&q=75 2x"
                style={
                  {
                    "color": "transparent",
                  }
                }
                width={250}
              />
            </div>
            <div
              className="sc-jrsKJM cQLIgn"
            >
              <img
                alt=""
                data-nimg="1"
                decoding="async"
                height={250}
                loading="lazy"
                onError={[Function]}
                onLoad={[Function]}
                src="/_next/image?url=%2Fimg%2FProject-Storybook-Desktop.png&w=750&q=75"
                srcSet="/_next/image?url=%2Fimg%2FProject-Storybook-Desktop.png&w=384&q=75 1x, /_next/image?url=%2Fimg%2FProject-Storybook-Desktop.png&w=750&q=75 2x"
                style={
                  {
                    "color": "transparent",
                  }
                }
                width={350}
              />
            </div>
          </div>
          <a
            background="transparent"
            border="none"
            borderRadius="0px"
            className="sc-gtsqUy jKDWbH"
            color="white"
            fontSize="16px"
            href="https://ricardosousabr.github.io/login-page-next/"
            padding="0px"
            textDecoration="none"
          >
            Storybook 
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
